parcels/field.py

```diff
@@ -1476,6 +1477,18 @@ class CField(Structure):
                 self.c_data_chunks[i] = None
 
         gridindexingtype_mapping = {"nemo": 0, "mitgcm": 1, "mom5": 2, "pop": 3}
+        interp_method_mapping = {
+            "LINEAR": 0,
+            "NEAREST": 1,
+            "CGRID_VELOCITY": 2,
+            "CGRID_TRACER": 3,
+            "BGRID_VELOCITY": 4,
+            "BGRID_W_VELOCITY": 5,
+            "BGRID_TRACER": 6,
+            "LINEAR_INVDIST_LAND_TRACER": 7,
+            "PARTIALSLIP": 8,
+            "FREESLIP": 9,
+        }
```

Comment:
This is still a bit ugly, because it mirrors the information from 
https://github.com/OceanParcels/Parcels/blob/997e3f86399cf319341a3d3f4fb1c5dc339e2580/parcels/include/index_search.h#L22-L30

It would be cleaner to either
1. Use the string here in the CField, instead of an integer. I tried using `c_char_p`, but that didn't work (yet)
2. Somehow import the typedefs from `index_search.h` (above) here
In any case, I think we want to avoid happing two places where these mappings between strings and integers occur...